### PR TITLE
Prevent adding to query details after unserve common has started

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR fixes the bug found in https://github.com/vitessio/vitess/issues/15685. The problem was that we were adding to the queryDetails even after unserveCommon had been called and after the kill had happened. This is troublesome and has been fixed by preventing addition to the query details right after `unserveCommon` starts.

## Related Issue(s)

- Fixes https://github.com/vitessio/vitess/issues/15685

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vttablet/tabletserver/livequeryz_test.go [hunk 1]
```diff
@@ -30,8 +30,8 @@ func TestLiveQueryzHandlerJSON(t *testing.T) {
 	req, _ := http.NewRequest("GET", "/livequeryz/?format=json", nil)
 
 	queryList := NewQueryList("test", sqlparser.NewTestParser())
-	queryList.Add(NewQueryDetail(context.Background(), &testConn{id: 1}))
-	queryList.Add(NewQueryDetail(context.Background(), &testConn{id: 2}))
+	_ = queryList.Add(NewQueryDetail(context.Background(), &testConn{id: 1}))
+	_ = queryList.Add(NewQueryDetail(context.Background(), &testConn{id: 2}))
```
Comment (dbussink) [inline go/vt/vttablet/tabletserver/livequeryz_test.go:34]: Do we want to instead `require.NoError` for the returns here? Would that help any future potential surprises if we also could return an error for other reasons?

**Review by dbussink — APPROVED** [review]

File: go/vt/vttablet/tabletserver/livequeryz_test.go (hunk 1 above)
Comment (GuptaManan100) [reply go/vt/vttablet/tabletserver/livequeryz_test.go]: Fixed

File: go/vt/vttablet/tabletserver/state_manager.go [hunk 2]
```diff
@@ -542,6 +542,8 @@ func (sm *stateManager) connect(tabletType topodatapb.TabletType) error {
 }
 
 func (sm *stateManager) unserveCommon() {
+	sm.markClusterAction(true)
+	defer sm.markClusterAction(false)
 	// We create a wait group that tracks whether all the queries have been terminated or not.
```
[the remark this answers is not in the document]
Comment (GuptaManan100) [reply go/vt/vttablet/tabletserver/state_manager.go]: Fixed

**Review by harshit-gangal — APPROVED** [review]

File: go/vt/vttablet/tabletserver/state_manager.go [hunk 3]
```diff
@@ -850,3 +854,10 @@ func (sm *stateManager) IsServingString() string {
 func (sm *stateManager) SetUnhealthyThreshold(v time.Duration) {
 	sm.unhealthyThreshold.Store(v.Nanoseconds())
 }
+
+// markClusterAction marks whether a cluster action is in progress or not for all the query details.
+func (sm *stateManager) markClusterAction(ca ClusterActionState) {
+	sm.statefulql.SetClusterAction(ca)
+	sm.statelessql.SetClusterAction(ca)
+	sm.olapql.SetClusterAction(ca)
+}
```
Comment (harshit-gangal) [inline go/vt/vttablet/tabletserver/state_manager.go:863]: over here we should ignore `ClusterActionNoQueries` when the state is `ClusterActionNotInProgress`

Comment (harshit-gangal) [reply go/vt/vttablet/tabletserver/state_manager.go:863]: I can see the check is in `func (ql *QueryList) SetClusterAction(ca ClusterActionState)`

**Pull request merged**